src/main/kotlin/ai/blindspot/ktoolz/extensions/OtherExtensions.kt

```diff
@@ -87,3 +87,70 @@ inline fun <T : Any> T.applyIf(shouldApply: Boolean, block: T.() -> Unit): T {
     return this
 }
 
+/**
+ * Shortens the string if needed. For example, returns "ABCD…" when called "ABCDEFHG.(5, "…")
+ */
+fun String.restrictLengthWithEllipsis(maxLength: Int, ellipsis: String = "…"): String {
+    if (this.length <= maxLength) return this
+    return this.substring(0, maxLength - ellipsis.length) + ellipsis
+}
+
+/**
+ * Creates a string like "className(details)", for example "Double(42.0)"
+ * Useful for implementing .toString() method (but in such case the first parameter MUST be passed).
+ */
+fun Any.toLongDebugString(description: String? = null, brackets: String = "()", className: String? = null): String {
```

Comment:
I will do as you say - I don't want to violate Blindspot's conventions.

My personal opinion is that
1. The name `toLongString` is not too descriptive about what the methods actually does.
2. I'd say that logging is part of debugging - we save logs in order to find bugs (that is, to debug) in what happens during the productive code. If the word "debug" means something else to various people, it might not be bad to try to use a less ambiguous word instead, if possible.
3. Even with your narrower definition of "debug", it can sometimes be useful to keep some debug methods in the codebase - only in cases when its potential future usefulness prevails the opposite considerations (just like you told me to keep my .toString overrides).

But I can be wrong.

Reply:
Just a quick response. 

1. At least for me, `toLongString` and `toLongDebugString` have exactly the same descriptive value. The word `debug` does not add any additional meaning to that method. E.g. in logging, `logger.debug {...}` has its meaning because you immediately know that you have to use `debug` level in config to see the output. In this case, there is no such additional information associated with that word.
1. Logs are used during debugging but not only for debugging. But I don't disagree, I just don't see additional value by adding the word `debug` in the method name
1. Again, I don't disagree, but in such a case there is no need to specifically mention it is a debug method. Just make it a normal part of the code